draft-ietf-quic-transport.md

```diff
@@ -712,23 +737,26 @@ sequence of frames, as described in {{frames}}.
 
 ## Connection ID {#connection-id}
 
-QUIC connections are identified by their 64-bit Connection ID.  All long headers
-contain a Connection ID.  Short headers indicate the presence of a Connection ID
-using the Omit Connection ID flag.  When present, the Connection ID is in the
-same location in all packet headers, making it straightforward for middleboxes,
-such as load balancers, to locate and use it.
+A connection ID is used to ensure consistent routing of packets.  The long
+header contains two connection IDs: the Destination Connection ID is chosen by
+the recipient of the packet and is used to provide consistent routing; the
+Source Connection ID is used to set the Destination Connection ID used by a
+peer.
 
-The client MUST choose a random connection ID and use it in Initial packets
-({{packet-initial}}) and 0-RTT packets ({{packet-protected}}).
+During the handshake, packets with the long header are used to establish the
+Destination Connection ID that each endpoint uses.  Each endpoint sends the
+value that they wish their peer to use in the Source Connection ID field.  Upon
+receiving a packet, each endpoint sets the Destination Connection ID to match
+the value of the Source Connection ID that they receive.
 
-When the server receives a Initial packet and decides to proceed with the
-handshake, it chooses a new value for the connection ID and sends that in a
-Retry ({{packet-retry}}) or Handshake ({{packet-handshake}}) packet.  The server
-MAY choose to use the value that the client initially selects.
+Short headers only include the Destination Connection ID and omit the explicit
+length.
 
-Once the client receives the connection ID that the server has chosen, it MUST
-use it for all subsequent Handshake ({{packet-handshake}}) and 1-RTT
-({{packet-protected}}) packets but not for 0-RTT packets ({{packet-protected}}).
+The very first packet sent by a client client includes a random value for
+Destination Connection ID.  The same value MUST be used for all 0-RTT packets
+sent on that connection ({{packet-protected}}).  This randomized value is used
+to determine the handshake packet protection keys (see Section 5.2.2 of
+{{QUIC-TLS}}).
 
 Server's Version Negotiation ({{packet-version}}) and Retry ({{packet-retry}})
 packets MUST use connection ID selected by the client.
```

Comment:
I know you didn't touch this, but it's missing a "the."  Also, should this say "both Source and Destination CIDs"?